Gordon360/Services/EventService.cs

```diff
@@ -27,223 +24,52 @@ namespace Gordon360.Services
 
         // Set the namespace for XML Paths
-        private XNamespace r25 = "http://www.collegenet.com/r25";
+        private readonly XNamespace r25 = "http://www.collegenet.com/r25";
 
         public EventService(IUnitOfWork unitOfWork)
         {
             _unitOfWork = unitOfWork;
         }
 
-
         /// <summary>
-        /// Helper function to set the route we are making our request to in 25Live
+        /// Access the memory stream created by the cached task and parse it into events
         /// </summary>
-        /// <param name="EventID"></param>
-        /// <param name="type"></param>
-        /// <returns></returns>
-        public string GetRoute(string EventID, string type) 
-        // all these '&event_type_id=<somId+someId>'s might need to change, actually we don't use the one that just gets a single type
+        /// <returns>All events for the current academic year.</returns>
+        public IEnumerable<EventViewModel> GetAllEvents()
         {
-            // Get the current year
-            string year = Helpers.GetFirstEventDate();
-            // Make an empty url string
-            string requestUrl = "";
-
-            // Return a list of all chapel events
-            if (EventID == "All")
-            {
-                // Set our api route and fill in the event information we would like
-                requestUrl = "https://25live.collegenet.com/25live/data/gordon/run/events.xml?/&event_type_id=14+19+28+57&state=2&end_after=" + year + "&scope=extended";
-            }
 
-            // If the type is "s", then it is a single event request, "m" is multiple event IDs
-            else if (type == "s" || type == "S" || type == "m" || type == "M")
-            {
-                // For 25Live, additional types must be separated by "+", so we replace the $ with "+"s here
-                if (type == "m" || type == "M")
-                {
-                    if (EventID.Contains('$'))
-                    {
-                        EventID = EventID.Replace('$', '+');
-                    }
-                }
-                // Set our api route and fill in the event information we would like
-                requestUrl = "https://25live.collegenet.com/25live/data/gordon/run/events.xml?/&event_id=" + EventID + "&scope=extended";
-
-            }
-            // If it is a type "t", then it is a search for a type"
-            // Same logic is used as above 
-            else if (type == "t" || type == "T")
-            {
-                if (EventID.Contains('$'))
-                {
-                    EventID = EventID.Replace('$', '+');
-                }
-                requestUrl = "https://25live.collegenet.com/25live/data/gordon/run/events.xml?/&event_type_id=" + EventID + "&state=2&end_after=" + year + "&scope=extended";
-            }
-
-            return requestUrl;
+            return Data.AllEvents.Descendants(r25 + "event").Select(e => new EventViewModel(e));
```

Comment:
The stark difference is actually mostly because GitHub got confused. If you look at the method signatures, you'll notice this one is actually GetAllEvents but the deleted code is the defunct helper method GetRoute. In the old version of the file, GetAllEvents is on line 126, after the two methods that got removed.